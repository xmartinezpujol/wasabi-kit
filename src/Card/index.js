import React from 'react';

import {
  Background,
  CardContainer,
  Image,
} from './styles';

function Card(props) {
  return (
    <CardContainer
      dimensions={props.dimensions}
      height={props.height}
      clickable={props.clickable}
      color={props.color}
      style={props.styleCont}
      onClick={props.onSelection}
    >
      {props.cover &&
        <Image
          url={props.cover}
          zoom={props.zoom}
          imgHeight={props.imgHeight}
          style={props.style}
        />
      }
      {!props.cover &&
        <Background color={props.color} style={props.style} />
      }
      {props.children}
    </CardContainer>
  );
}

export default Card;
