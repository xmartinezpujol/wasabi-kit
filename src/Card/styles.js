import styled from 'styled-components';

import COLOR_PALETTE from '../Constants';

const CardContainer = styled.div(
  {
    textDecoration: 'none',
    fontSize: 16,
    margin: 10,
    border: 'none',
    display: 'inline-block',
    padding: '10px 20px',
    transition: '0.25s cubic-bezier(0.17, 0.67, 0.52, 0.97)',
    borderRadius: 5,
    color: '#000',
    boxShadow: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
    ':focus': { outline: 0 },
  },
  props => ({
    backgroundColor: COLOR_PALETTE[props.color],
    height: props.height,
    cursor: props.clickable ? 'pointer' : 'initial',
    ':hover': {
      opacity: props.clickable ? 0.8 : 1,
      transform: props.clickable ? 'translateY(-1px)' : 'none',
      boxShadow: props.clickable ? '0 4px 6px rgba(50,50,93,.31), 0 1px 2px rgba(0,0,0,.40)' : '0 4px 6px rgba(50,50,93,.11), 0 1px 2px rgba(0,0,0,.20)',
    },
    ':active': {
      transform: props.clickable ? 'translateY(1px)' : 'none',
    },
    '@media(min-width: 1200px)': {
      minWidth: props.dimensions.lg.w,
      maxWidth: props.dimensions.lg.w,
    },
    '@media(max-width: 1200px)': {
      minWidth: props.dimensions.md.w,
      maxWidth: props.dimensions.md.w,
    },
    '@media(max-width: 992px)': {
      minWidth: props.dimensions.sm.w,
      maxWidth: props.dimensions.sm.w,
    },
    '@media(max-width: 768px)': {
      minWidth: props.dimensions.xs.w,
      maxWidth: props.dimensions.xs.w,
      margin: '10px auto',
    },
  }),
);

const Background = styled.div(
  {
    height: '140px',
    margin: '-10px -20px 10px',
  },
  props => (
    {
      background: props.color,
    }
  ),
);

const Image = styled.div(
  {
    margin: '-10px -20px 10px',
    backgroundSize: '100%',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: '50% 50%',
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    transition: 'all 0.4s ease',
  },
  props => ({
    height: props.imgHeight.lg,
    backgroundImage: `url(${props.url})`,
    ':hover': {
      backgroundSize: props.zoom ? '120%' : '100%',
    },
    '@media(max-width: 1200px)': {
      height: props.imgHeight.md,
    },
    '@media(max-width: 992px)': {
      height: props.imgHeight.sm,
    },
    '@media(max-width: 767px)': {
      height: props.imgHeight.xs,
    },
  }),
);

export {
  Background,
  CardContainer,
  Image,
};
