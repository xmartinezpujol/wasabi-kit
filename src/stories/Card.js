import React from 'react';
import styled from 'styled-components';

import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, object } from '@storybook/addon-knobs';

import Text from '../Text';
import Card from '../Card';
import View from '../View';

const dimensions = object('Dimensions', {
  xs: { w: 310, h: 240 },
  sm: { w: 385, h: 285 },
  md: { w: 385, h: 285 },
  lg: { w: 385, h: 285 }
});
const imgHeights = object('Image heights', {
  xs: 130,
  sm: 165,
  md: 165,
  lg: 165
});

const dimensionsSquare = object('Dimensions', {
  xs: { w: '100%', h: 330 },
  sm: { w: 'calc(50% - 20px)', h: 330 },
  md: { w: 'calc(33% - 17px)', h: 330 },
  lg: { w: 'calc(33% - 17px)', h: 330 }
});
const imgHeightsSquare = object('Image heights', {
  xs: 330,
  sm: 330,
  md: 330,
  lg: 330
});

const ViewAdapter = styled(View)({
  flexDirection: 'row',
  flexWrap: 'wrap',
  '@media(max-width: 768px)': {
    flexDirection: 'column'
  }
});

storiesOf('Cards', module)
  .addDecorator(withKnobs)
  .add('img + text', () => (
    <View padding={100} container type="purewhite">
      <Card
        cover={text(
          'Cover',
          'https://upload.wikimedia.org/wikipedia/commons/1/1e/Indiandishes.jpg'
        )}
        color="rice"
        onClick={action('clicked')}
        clickable={boolean('Clickable', false)}
        zoom={boolean('Zoom', false)}
        dimensions={dimensions}
        imgHeight={imgHeights}
      >
        <Text type="h2">{text('Title', 'Curry')}</Text>
        <Text type="p1">
          {text(
            'Text',
            'Curry (/ˈkʌri/, sometimes /ˈkɜːri/, plural curries) is an umbrella term referring to a number of dishes originating in the cuisine of the Indian subcontinent. .'
          )}
        </Text>
      </Card>
    </View>
  ))
  .add('img', () => (
    <ViewAdapter container>
      <Card
        cover={text(
          'Cover',
          'https://upload.wikimedia.org/wikipedia/commons/1/1e/Indiandishes.jpg'
        )}
        color="rice"
        onClick={action('clicked')}
        clickable={boolean('Clickable', true)}
        zoom={boolean('Zoom', true)}
        dimensions={dimensionsSquare}
        imgHeight={imgHeightsSquare}
        style={{
          margin: 0,
          padding: 0,
          borderRadius: 5
        }}
        styleCont={{ padding: 0 }}
      />
      <Card
        cover={text(
          'Cover',
          'https://upload.wikimedia.org/wikipedia/commons/1/1e/Indiandishes.jpg'
        )}
        color="rice"
        onClick={action('clicked')}
        clickable={boolean('Clickable', true)}
        zoom={boolean('Zoom', true)}
        dimensions={dimensionsSquare}
        imgHeight={imgHeightsSquare}
        style={{
          margin: 0,
          padding: 0,
          borderRadius: 5
        }}
        styleCont={{ padding: 0 }}
      />
      <Card
        cover={text(
          'Cover',
          'https://upload.wikimedia.org/wikipedia/commons/1/1e/Indiandishes.jpg'
        )}
        color="rice"
        onClick={action('clicked')}
        clickable={boolean('Clickable', true)}
        zoom={boolean('Zoom', true)}
        dimensions={dimensionsSquare}
        imgHeight={imgHeightsSquare}
        style={{
          margin: '0px auto',
          padding: 0,
          borderRadius: 5
        }}
        styleCont={{ padding: 0 }}
      />
    </ViewAdapter>
  ));
