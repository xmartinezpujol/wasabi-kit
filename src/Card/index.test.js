import React from 'react';
import { shallow } from 'enzyme';
import serializer, { toHaveStyleRule } from 'jest-styled-components';

import Card from './';

const dimensionsSquare = {
  xs: { w: '100%', h: 330 },
  sm: { w: 'calc(50% - 20px)', h: 330 },
  md: { w: 'calc(33% - 17px)', h: 330 },
  lg: { w: 'calc(33% - 17px)', h: 330 },
};
const imgHeightsSquare = {
  xs: 330,
  sm: 330,
  md: 330,
  lg: 330,
};

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card
        cover="https://upload.wikimedia.org/wikipedia/commons/1/1e/Indiandishes.jpg"
        color="white"
        onClick={() => console.log('clicked')}
        clickable
        dimensions={dimensionsSquare}
        imgHeight={imgHeightsSquare}
        style={{
          margin: 0,
          padding: 0,
          backgroundSize: 'cover',
          borderRadius: 5,
        }}
        styleCont={{ padding: 0 }}
      />);
  });
  it('renders correctly', () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
});
