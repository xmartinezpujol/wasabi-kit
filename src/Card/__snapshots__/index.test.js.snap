// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "styles__CardContainer-sc-1ylq150-0 HsBeW",
      "color": "white",
      "style": "padding:0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "styles__Image-sc-1ylq150-2 jqhOZa",
          "style": "margin:0;padding:0;background-size:cover;border-radius:5px",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "color": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "color": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
